### PR TITLE
Pagination

### PR DESCRIPTION
**Pull request by gutakk**
## What happened
- Add pagination to keywords list page

 
## Insight
- Use `Kaminari` to generate pagination
- Use `Kaminari` bootstrap 4 theme for pagination styling
- Limit 25 keywords per page 

## Proof Of Work
- Does not show pagination when keywords less than 25
Screen Shot 2563-10-20 at 18 55 29

- Show 3 page when upload 60 keywords csv file
Screen Shot 2563-10-20 at 18 56 49

- Last page show only 10 keywords
Screen Shot 2563-10-20 at 18 57 22

- Does not show keyword table when navigate to out of range page `(http://localhost:3000/keywords?page=4)`
Screen Shot 2563-10-20 at 18 51 35

### CODE REVIEW

File: spec/systems/pagination_spec.rb [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+      visit keywords_path
+
+      within 'form' do
+        fill_in('username', with: user[:username])
+        fill_in('password', with: 'password')
+
+        click_button(I18n.t('auth.login'))
+      end
+
+      attach_file('csv_import_form[file]', file_path)
```
Comment (longnd) [inline spec/systems/pagination_spec.rb]: similar to [this comment](https://github.com/gutakk/ruby-google-scraper/pull/11#discussion_r507513915), should we create the articles in DB directly instead?

File: app/assets/stylesheets/components/_pagination.scss [hunk 2]
```diff
@@ -0,0 +1,14 @@
+.pagination {
+  margin-bottom: 3rem;
+
+  .page-link {
```
Comment (longnd) [inline app/assets/stylesheets/components/_pagination.scss:4]: is `link` an element of the `.page` block, if yes, should we name it to follow the [convention on Compass](https://compass.nimblehq.co/development/code-conventions/css/#class-names)

```css
   .page__link {
   }
   
   .page__item {
    }
```

Comment (gutakk) [reply app/assets/stylesheets/components/_pagination.scss:4]: It is not, it is the bootstrap class name.

File: spec/systems/pagination_spec.rb (hunk 1 above)
Comment (gutakk) [reply spec/systems/pagination_spec.rb]: Fixed in e265c9410da9c88c674f47280fb981fb94f20dbd

File: app/views/keywords/index.html.slim [hunk 3]
```diff
@@ -4,14 +4,17 @@
 
   = render(partial: 'search')
 
-  - if keywords.present?
-    table class="table table-hover my-5 w-50"
+  - if keywords.present?
```
Comment (andyduong1920) [inline app/views/keywords/index.html.slim]: please remove empty spacea after `present?`

File: spec/systems/pagination_spec.rb [hunk 4]
```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe 'pagination', type: :system do
+  ActiveJob::Base.queue_adapter = :test
```
Comment (andyduong1920) [inline spec/systems/pagination_spec.rb]: 🤔

File: app/views/keywords/index.html.slim (hunk 3 above)
Comment (gutakk) [reply app/views/keywords/index.html.slim]: Fixed in d5684d16e6f21c255cc669a83245943ec64767cb

File: spec/systems/pagination_spec.rb (hunk 4 above)
Comment (gutakk) [reply spec/systems/pagination_spec.rb]: Fixed in 1a828031b2751651923d865f6120b402ecc7dd5b

**Review by andyduong1920 — APPROVED** [review]

**Review by andyduong1920 — APPROVED** [review]

**Review by longnd — APPROVED** [review]

**Pull request merged**